### PR TITLE
fix HybridRelations::belongsToMany not compatible with Model::belongsToMany 

### PR DESCRIPTION
**Pull request by akaroot**
This should fix issue #1273 for Laravel 5.5.
This is incompatible with Laravel < 5.5 so it couldn't be used for v 3.2.*

### CODE REVIEW

File: src/Jenssegers/Mongodb/Eloquent/HybridRelations.php [hunk 1]
```diff
@@ -204,17 +204,20 @@ public function morphTo($name = null, $type = null, $id = null)
         }
     }
 
-    /**
```
Comment (linhc1994) [inline src/Jenssegers/Mongodb/Eloquent/HybridRelations.php]: Command syntax error

**Comment by jenssegers** [comment]
Looks like there are more things broken according to travis.

**Comment by nshelms** [comment]
I went through the file and updated all of the functions while keeping the composer.json changes as well. Here's a patch file. Can someone tell me if it fixes this?

[patch--170906.txt](https://github.com/jenssegers/laravel-mongodb/files/1283014/patch--170906.txt)

**Comment by godfreymakori** [comment]
Check out this gist https://github.com/jenssegers/laravel-mongodb/pull/1282/commits/066e2193bc50d2dce1802c00d46ce3b7d252a1c6

**Comment by godfreymakori** [comment]
Check out this gist https://github.com/jenssegers/laravel-mongodb/pull/1282/commits/066e2193bc50d2dce1802c00d46ce3b7d252a1c6

**Comment by moiz2050** [comment]
@jenssegers i am still having this issue while installing version 3.2.3 will L5.5

`  [ErrorException]
  Declaration of Jenssegers\Mongodb\Eloquent\HybridRelations::belongsToMany($
  related, $collection = NULL, $foreignKey = NULL, $otherKey = NULL, $relatio
  n = NULL) should be compatible with Illuminate\Database\Eloquent\Model::bel
  ongsToMany($related, $table = NULL, $foreignPivotKey = NULL, $relatedPivotK
  ey = NULL, $parentKey = NULL, $relatedKey = NULL, $relation = NULL)`

File: src/Jenssegers/Mongodb/Eloquent/HybridRelations.php [hunk 2]
```diff
@@ -204,17 +204,20 @@ public function morphTo($name = null, $type = null, $id = null)
...
+    **
```
Comment (alicompu) [inline src/Jenssegers/Mongodb/Eloquent/HybridRelations.php]: /

Comment (alicompu) [reply src/Jenssegers/Mongodb/Eloquent/HybridRelations.php]: miss the slash.

**Comment by divine** [comment]
Should be closed, https://github.com/jenssegers/laravel-mongodb/pull/1300 already merged

cc @Smolevich

**Pull request closed without merging**